src/libraries/System.Runtime/tests/System/DoubleTests.cs

```diff
@@ -272,6 +272,11 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
             yield return new object[] { (567.89).ToString(), defaultStyle, null, 567.89 };
             yield return new object[] { (-567.89).ToString(), defaultStyle, null, -567.89 };
             yield return new object[] { "1E23", defaultStyle, null, 1E23 };
+            yield return new object[] { "9007199254740997.0", defaultStyle, invariantFormat, 9007199254740996 };
+            yield return new object[] { "5.005", defaultStyle, invariantFormat, 5.005 };
+            yield return new object[] { "5.050", defaultStyle, invariantFormat, 5.05 };
+            yield return new object[] { "5005.0", defaultStyle, invariantFormat, 5005 };
+            yield return new object[] { "50050.0", defaultStyle, invariantFormat, 50050 };
```

Comment:
Just for my own interest, what were these four returning before the fix?

-- They likewise should likely have an explicit `.0` at the end, to ensure they are treated as `double` and not `int`

Reply:
All of them returned the correct values. I added them as test cases now just for thoroughness.